File: ListasAtiv.HASKELL/Questao5.hs
--  Defina uma função que retorna todos os números 
-- que são primos de um intervalo.

verificaSePrimo :: Int -> Bool
verificaSePrimo 1 = False
verificaSePrimo 2 = True
verificaSePrimo n 
    | (length [x | x <- [2 .. n-1], mod n x == 0]) > 0 = False
	| otherwise = True


criaListaPrimos lista = 
    [x | x <-lista, verificaSePrimo(x) == True]

main = do
    -- print(verificaSePrimo 6 :: Bool)
    let lista = [1..5]
    print(criaListaPrimos lista)

File: ListasAtiv.PYTHON/Questao5.py
def verificaSePrimo(num):
    if num < 2:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True


def criaListaPrimos(lista):
    listaPrimos = [x  for x in lista if verificaSePrimo(x)]
    return listaPrimos

lista = list(range(0,10))
print(criaListaPrimos(lista))


